### DISCUSSION TITLE
el.onclick=func, el.addEventListner('click',func)

### DISCUSSION THREAD

**Discussion by javascriptbiggosoo** [wanted-9th-3team/frontend-study#81] [프론트 기술 질문 공간]
이벤트 핸들러를 달아주는 두 방식
`el.onclick=func`, `el.addEventListner('click',func)`
에 차이가 있다면 무엇이 있는지 알려주세요.

**Comment by michoball** [comment]
onclick 특정 el를 클릭할 때 이벤트를 설정할 수 있고, html 속성으로 사용할 수 있습니다.  하지만 다른 이벤트를 추가적으로 달려고 하면 이벤트가 덮어씌워집니다. 
즉 el 당 하나의 이벤트만 연결할 수 있습니다. 

하지만 addEventListener('click', fn() )은 해당 el에  여러개의 이벤트를 달 수 있고 이벤트 버블링 단계 캡쳐링 단계를 구분지어 이벤트를 달 수도 있습니다.  다만 구버전 IE에서는 사용이 안되고 html 속성으로 사용할 수 없어 html 에서 사용하려면 <script> 태그 안에서 사용해야 합니다.

**Comment by WooGie911** [comment]
onclick 은 하나의 콜백함수(리스너)만 지정 할 수 있지만 addEventListner 는 여러 개의 이벤트 리스너를 추가 할 수 있습니다.
하지만 onclick 은 모든 브라우저와 버젼에서 호환되지만 addEventListner 는 IE 6,7,8에서는 호환되지 않습니다.
따라서 호환성을 생각한다면 onclick 을 사용하는 것이 맞겠지만, 그 외에는 addEventListner 를 사용하는 것이 좋다고 생각합니다.

**Comment by javascriptbiggosoo** [comment]
addEventListner('click',func)는 이벤트핸들러를 여러개 계속 더해줄수 있고, 세번째 인자로 버블링 캡처링 조정할 수 있습니다.
